include/conversation.php

```diff
@@ -467,24 +469,27 @@ function item_fieldlists() {
 
 /**
  * @brief SQL join for contacts that are needed for displaying items
+ *
+ * @param int $uid user id
  */
-function item_joins() {
+function item_joins($uid = 0) {
 	return sprintf("STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`contact-id`
 		AND NOT `contact`.`blocked`
 		AND ((NOT `contact`.`readonly` AND NOT `contact`.`pending` AND (`contact`.`rel` IN (%s, %s)))
 		OR `contact`.`self` OR (`item`.`id` != `item`.`parent`) OR `contact`.`uid` = 0)
 		INNER JOIN `contact` AS `author` ON `author`.`id`=`item`.`author-id` AND NOT `author`.`blocked`
 		INNER JOIN `contact` AS `owner` ON `owner`.`id`=`item`.`owner-id` AND NOT `owner`.`blocked`
+		LEFT JOIN `user-item` ON `user-item`.`iid` = `item`.`id` AND `user-item`.`uid` = %d
 		LEFT JOIN `event` ON `event-id` = `event`.`id`",
-		CONTACT_IS_SHARING, CONTACT_IS_FRIEND
+		CONTACT_IS_SHARING, CONTACT_IS_FRIEND, intval($uid)
 	);
 }
 
 /**
  * @brief SQL condition for items that are needed for displaying items
  */
 function item_condition() {
-	return "`item`.`visible` AND NOT `item`.`deleted` AND NOT `item`.`moderated`";
+	return "`item`.`visible` AND NOT `item`.`deleted` AND NOT `item`.`moderated` AND (`user-item`.`hidden` IS NULL OR NOT `user-item`.`hidden`) ";
```

Comment:
Per table definition, `user-item.hidden` can't be null.

Reply:
It is "null" whenever there is no "user-item" entry for this item id.

Reply:
I realized later, can you clarify the table usage?

Reply:
I don't understand. What do you mean?

Reply:
I pieced back its usage from the code, it enables a user to hide a public item from their specific view.

Reply:
I guess it's too late now. I don't understand anything. What does "I pieced back its usage from the code" mean?